### PR TITLE
Fix attempt to send unsupported message (13) via cap eth/67

### LINKED ISSUE #4731: Attempt to send unsupported message (13) via cap eth/67
Message from @fab-10 :

> investigating the issues  I found these error logs java.lang.UnsupportedOperationException: Attempt to send unsupported message (13) via cap eth/67

```
java.lang.UnsupportedOperationException: Attempt to send unsupported message (13) via cap eth/67
	at org.hyperledger.besu.ethereum.p2p.rlpx.connections.AbstractPeerConnection.doSend(AbstractPeerConnection.java:97)
	at org.hyperledger.besu.ethereum.p2p.rlpx.connections.AbstractPeerConnection.send(AbstractPeerConnection.java:86)
	at org.hyperledger.besu.ethereum.p2p.rlpx.connections.PeerConnection.sendForProtocol(PeerConnection.java:72)
	at org.hyperledger.besu.ethereum.eth.manager.EthPeer.lambda$sendRequest$5(EthPeer.java:361)
	at org.hyperledger.besu.ethereum.eth.manager.RequestManager.dispatchRequest(RequestManager.java:68)
	at org.hyperledger.besu.ethereum.eth.manager.EthPeer.sendRequest(EthPeer.java:360)
	at org.hyperledger.besu.ethereum.eth.manager.EthPeer.getNodeData(EthPeer.java:306)
	at org.hyperledger.besu.ethereum.eth.manager.task.GetNodeDataFromPeerTask.lambda$sendRequest$0(GetNodeDataFromPeerTask.java:70)
	at org.hyperledger.besu.ethereum.eth.manager.PendingPeerRequest.sendRequest(PendingPeerRequest.java:74)
	at java.base/java.util.Optional.ifPresent(Optional.java:183)
	at org.hyperledger.besu.ethereum.eth.manager.PendingPeerRequest.attemptExecution(PendingPeerRequest.java:65)
	at org.hyperledger.besu.ethereum.eth.manager.EthPeers.executePeerRequest(EthPeers.java:151)
	at org.hyperledger.besu.ethereum.eth.manager.task.AbstractPeerRequestTask.sendRequestToPeer(AbstractPeerRequestTask.java:94)
	at org.hyperledger.besu.ethereum.eth.manager.task.GetNodeDataFromPeerTask.sendRequest(GetNodeDataFromPeerTask.java:67)
	at org.hyperledger.besu.ethereum.eth.manager.task.AbstractPeerRequestTask.executeTask(AbstractPeerRequestTask.java:59)
	at org.hyperledger.besu.ethereum.eth.manager.task.AbstractEthTask.executeTaskTimed(AbstractEthTask.java:150)
	at org.hyperledger.besu.ethereum.eth.manager.task.AbstractEthTask.run(AbstractEthTask.java:75)
	at org.hyperledger.besu.ethereum.eth.manager.task.AbstractEthTask.executeSubTask(AbstractEthTask.java:120)
	at org.hyperledger.besu.ethereum.eth.manager.task.RetryingGetNodeDataFromPeerTask.executePeerTask(RetryingGetNodeDataFromPeerTask.java:64)
	at org.hyperledger.besu.ethereum.eth.manager.task.AbstractRetryingPeerTask.executeTask(AbstractRetryingPeerTask.java:90)
	at org.hyperledger.besu.ethereum.eth.manager.task.AbstractEthTask.executeTaskTimed(AbstractEthTask.java:150)
	at org.hyperledger.besu.ethereum.eth.manager.task.AbstractEthTask.run(AbstractEthTask.java:75)
	at org.hyperledger.besu.ethereum.eth.sync.worldstate.WorldStatePeerTrieNodeFinder.findByGetNodeData(WorldStatePeerTrieNodeFinder.java:115)
	at org.hyperledger.besu.ethereum.eth.sync.worldstate.WorldStatePeerTrieNodeFinder.getAccountStorageTrieNode(WorldStatePeerTrieNodeFinder.java:93)
	at org.hyperledger.besu.ethereum.bonsai.BonsaiWorldStateKeyValueStorage.lambda$getAccountStorageTrieNode$4(BonsaiWorldStateKeyValueStorage.java:158)
	at java.base/java.util.Optional.flatMap(Optional.java:294)
	at org.hyperledger.besu.ethereum.bonsai.BonsaiWorldStateKeyValueStorage.lambda$getAccountStorageTrieNode$5(BonsaiWorldStateKeyValueStorage.java:156)
	at java.base/java.util.Optional.or(Optional.java:318)
	at org.hyperledger.besu.ethereum.bonsai.BonsaiWorldStateKeyValueStorage.getAccountStorageTrieNode(BonsaiWorldStateKeyValueStorage.java:154)
	at org.hyperledger.besu.ethereum.bonsai.BonsaiWorldStateKeyValueStorage.lambda$getStorageValueBySlotHash$6(BonsaiWorldStateKeyValueStorage.java:196)
	at org.hyperledger.besu.ethereum.trie.StoredNodeFactory.retrieve(StoredNodeFactory.java:96)
	at org.hyperledger.besu.ethereum.trie.StoredNode.load(StoredNode.java:130)
	at org.hyperledger.besu.ethereum.trie.StoredNode.accept(StoredNode.java:63)
	at org.hyperledger.besu.ethereum.trie.StoredMerklePatriciaTrie.get(StoredMerklePatriciaTrie.java:119)
	at org.hyperledger.besu.ethereum.bonsai.BonsaiWorldStateKeyValueStorage.getStorageValueBySlotHash(BonsaiWorldStateKeyValueStorage.java:200)
	at org.hyperledger.besu.ethereum.bonsai.BonsaiPersistedWorldState.getStorageValueBySlotHash(BonsaiPersistedWorldState.java:404)
	at org.hyperledger.besu.ethereum.bonsai.BonsaiWorldStateUpdater.getStorageValueBySlotHash(BonsaiWorldStateUpdater.java:332)
	at org.hyperledger.besu.ethereum.bonsai.BonsaiWorldStateUpdater.getStorageValue(BonsaiWorldStateUpdater.java:319)
	at org.hyperledger.besu.ethereum.bonsai.BonsaiAccount.getStorageValue(BonsaiAccount.java:208)
	at org.hyperledger.besu.evm.worldstate.UpdateTrackingAccount.getStorageValue(UpdateTrackingAccount.java:213)
	at org.hyperledger.besu.evm.worldstate.UpdateTrackingAccount.getStorageValue(UpdateTrackingAccount.java:213)
	at org.hyperledger.besu.evm.worldstate.UpdateTrackingAccount.getStorageValue(UpdateTrackingAccount.java:213)
	at org.hyperledger.besu.evm.operation.SLoadOperation.execute(SLoadOperation.java:58)
	at org.hyperledger.besu.evm.EVM.runToHalt(EVM.java:282)
	at org.hyperledger.besu.evm.processor.AbstractMessageProcessor.codeExecute(AbstractMessageProcessor.java:161)
	at org.hyperledger.besu.evm.processor.AbstractMessageProcessor.process(AbstractMessageProcessor.java:173)
	at org.hyperledger.besu.ethereum.mainnet.MainnetTransactionProcessor.process(MainnetTransactionProcessor.java:492)
	at org.hyperledger.besu.ethereum.mainnet.MainnetTransactionProcessor.processTransaction(MainnetTransactionProcessor.java:404)
	at org.hyperledger.besu.ethereum.mainnet.AbstractBlockProcessor.processBlock(AbstractBlockProcessor.java:104)
	at org.hyperledger.besu.ethereum.mainnet.BlockProcessor.processBlock(BlockProcessor.java:76)
	at org.hyperledger.besu.ethereum.MainnetBlockValidator.processBlock(MainnetBlockValidator.java:205)
	at org.hyperledger.besu.ethereum.MainnetBlockValidator.validateAndProcessBlock(MainnetBlockValidator.java:135)
	at org.hyperledger.besu.ethereum.MainnetBlockValidator.validateAndProcessBlock(MainnetBlockValidator.java:74)
	at org.hyperledger.besu.ethereum.eth.sync.backwardsync.BackwardSyncContext.saveBlock(BackwardSyncContext.java:320)
	at org.hyperledger.besu.ethereum.eth.sync.backwardsync.ForwardSyncStep.saveBlocks(ForwardSyncStep.java:124)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:642)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2073)
	at org.hyperledger.besu.ethereum.eth.manager.task.RetryingGetBlocksFromPeersTask.lambda$executeTaskOnCurrentPeer$0(RetryingGetBlocksFromPeersTask.java:88)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:642)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2073)
	at org.hyperledger.besu.ethereum.eth.manager.task.AbstractPeerRequestTask.lambda$executeTask$2(AbstractPeerRequestTask.java:83)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:859)
	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:837)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2073)
	at org.hyperledger.besu.ethereum.eth.manager.task.AbstractPeerRequestTask.lambda$handleMessage$3(AbstractPeerRequestTask.java:110)
	at java.base/java.util.Optional.ifPresent(Optional.java:183)
	at org.hyperledger.besu.ethereum.eth.manager.task.AbstractPeerRequestTask.handleMessage(AbstractPeerRequestTask.java:108)
	at org.hyperledger.besu.ethereum.eth.manager.task.AbstractPeerRequestTask.lambda$executeTask$0(AbstractPeerRequestTask.java:67)
	at org.hyperledger.besu.ethereum.eth.manager.RequestManager$ResponseStream.dispatchBufferedResponses(RequestManager.java:211)
	at org.hyperledger.besu.ethereum.eth.manager.RequestManager$ResponseStream.processMessage(RequestManager.java:202)
	at org.hyperledger.besu.ethereum.eth.manager.RequestManager.lambda$dispatchResponse$0(RequestManager.java:82)
	at java.base/java.util.Optional.ifPresentOrElse(Optional.java:201)
	at org.hyperledger.besu.ethereum.eth.manager.RequestManager.dispatchResponse(RequestManager.java:81)
	at org.hyperledger.besu.ethereum.eth.manager.EthPeer.lambda$dispatch$7(EthPeer.java:386)
	at java.base/java.util.Optional.ifPresentOrElse(Optional.java:201)
	at org.hyperledger.besu.ethereum.eth.manager.EthPeer.dispatch(EthPeer.java:385)
	at org.hyperledger.besu.ethereum.eth.manager.EthPeers.dispatchMessage(EthPeers.java:160)
	at org.hyperledger.besu.ethereum.eth.manager.EthProtocolManager.processMessage(EthProtocolManager.java:300)
	at org.hyperledger.besu.ethereum.p2p.network.NetworkRunner.lambda$setupHandlers$0(NetworkRunner.java:135)
	at org.hyperledger.besu.ethereum.p2p.rlpx.connections.PeerConnectionEvents.lambda$dispatchMessage$1(PeerConnectionEvents.java:64)
	at org.hyperledger.besu.util.Subscribers.lambda$forEach$0(Subscribers.java:112)
	at java.base/java.lang.Iterable.forEach(Iterable.java:75)
	at org.hyperledger.besu.util.Subscribers.forEach(Subscribers.java:109)
	at org.hyperledger.besu.ethereum.p2p.rlpx.connections.PeerConnectionEvents.dispatchMessage(PeerConnectionEvents.java:64)
	at org.hyperledger.besu.ethereum.p2p.rlpx.connections.netty.ApiHandler.channelRead0(ApiHandler.java:99)
	at org.hyperledger.besu.ethereum.p2p.rlpx.connections.netty.ApiHandler.channelRead0(ApiHandler.java:33)
	at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:286)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:327)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:299)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:722)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:658)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:584)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:496)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

### PR DESCRIPTION
**Pull request by Gabriel-Trintinalia**
## PR description
WorldStatePeerTrieNodeFinder - Do not call GetNodeData if the protocol version is greater than Eth66

## Fixed Issue(s)
fixes #4731

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if
    [updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).

## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: ethereum/eth/src/test/java/org/hyperledger/besu/ethereum/eth/sync/worldstate/WorldStatePeerTrieNodeFinderTest.java [hunk 1]
```diff
@@ -258,4 +272,36 @@ public void getAccountStorageTrieNodeShouldReturnEmptyWhenFoundNothing() {
...
+    when(ethProtocolManager.getHighestProtocolVersion()).thenReturn(EthProtocolVersion.V66);
+
+    worldStatePeerTrieNodeFinder.getAccountStorageTrieNode(
+        Hash.wrap(Bytes32.random()), Bytes.EMPTY, Hash.wrap(Bytes32.random()));
+
+    // assert findByGetNodeData is called once
+    verify(worldStatePeerTrieNodeFinder, times(1)).findByGetNodeData(any());
+
+    // assert findByGetTrieNodeData is called once
+    verify(worldStatePeerTrieNodeFinder, times(1)).findByGetTrieNodeData(any(), any(), any());
```
[the remark this answers is not in the document]
Comment (Gabriel-Trintinalia) [reply ethereum/eth/src/test/java/org/hyperledger/besu/ethereum/eth/sync/worldstate/WorldStatePeerTrieNodeFinderTest.java:289]: The code first tries findByGetNodeData and falls back to findByGetTrieNodeData if nothing is found. 
With this PR, if Eth67, only findByGetTrieNodeData is used. If Eth66, the previous behaviour is still valid.

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/worldstate/WorldStatePeerTrieNodeFinder.java [hunk 2]
```diff
@@ -90,7 +98,10 @@ public Optional<Bytes> getAccountStorageTrieNode(
       return cachedValue;
     }
     final Optional<Bytes> response =
-        findByGetNodeData(Hash.wrap(nodeHash))
+        // Call findByGetNodeData only if protocol version < eth 67
+        (protocolManager.getHighestProtocolVersion() < EthProtocolVersion.V67
+                ? findByGetNodeData(Hash.wrap(nodeHash))
```
Comment (jframe) [inline ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/worldstate/WorldStatePeerTrieNodeFinder.java:103]: Does the Fastsync RequestDataStep also need such a check? It also uses GetNodeData https://github.com/hyperledger/besu/blob/main/ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/worldstate/RequestDataStep.java#L48

**Review by jframe — APPROVED** [review]
LGTM

File: ethereum/eth/src/test/java/org/hyperledger/besu/ethereum/eth/sync/worldstate/WorldStatePeerTrieNodeFinderTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (matkt) [reply ethereum/eth/src/test/java/org/hyperledger/besu/ethereum/eth/sync/worldstate/WorldStatePeerTrieNodeFinderTest.java:289]: seems ok to me

**Review by matkt — APPROVED** [review]

**Pull request merged**